src/main/java/org/openstreetmap/atlas/checks/validation/tag/HighwayMissingNameAndRefTagCheck.java

```diff
@@ -0,0 +1,241 @@
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Function calculate angle between Edges
+     * 
+     * @param edge1
+     *            an edge
+     * @param edge2
+     *            another edge
+     * @return Angle between edges
+     */
+    private Angle angleDiffBetweenEdges(final Edge edge1, final Edge edge2)
+    {
+        final Optional<Heading> edge1heading = edge1.asPolyLine().finalHeading();
+        final Optional<Heading> edge2heading = edge2.asPolyLine().initialHeading();
+        if (edge1heading.isPresent() && edge2heading.isPresent())
+        {
+            return edge1heading.get().difference(edge2heading.get());
+        }
+        return Angle.NONE;
+    }
+
+    /**
+     * Function to collect osm name tags
+     * 
+     * @param edges
+     *            a set of edges (connected edges from edge in question)
+     * @return set of strings representing name tags
+     */
+    private Set<String> collectEdgeNameTags(final Set<Edge> edges)
+    {
+        final Set<String> edgeNames = new HashSet<>();
+        for (final Edge edge : edges)
+        {
+            final Map<String, String> tags = edge.getTags();
+            if (tags.containsKey(NameTag.KEY))
+            {
+                edgeNames.add(tags.get(NameTag.KEY));
+            }
+        }
+        return edgeNames;
+    }
+
+    /**
+     * Function to collect osm reference tags
+     * 
+     * @param edges
+     *            a set of edges (connected edges from edge in question)
+     * @return Set of strings representing reference tags
+     */
+    private Set<String> collectEdgeRefTags(final Set<Edge> edges)
+    {
+        final Set<String> edgeRefs = new HashSet<>();
+        for (final Edge edge : edges)
+        {
+            final Map<String, String> tags = edge.getTags();
+            if (tags.containsKey(ReferenceTag.KEY))
+            {
+                edgeRefs.add(tags.get(ReferenceTag.KEY));
+            }
+        }
+        return edgeRefs;
+    }
+
+    /**
+     * Determines if Edge object is missing both name and ref tag
+     * 
+     * @param tags
+     *            object tags
+     * @return boolean if Edge does not contain either name or ref tag.
+     */
+    private boolean highwayMissingBothNameAndRefTag(final Map<String, String> tags)
+    {
+        return !tags.containsKey(NameTag.KEY.toLowerCase())
+                && !tags.containsKey(ReferenceTag.KEY.toLowerCase());
+    }
+
+    /**
+     * Function to determine if edge is a small connector edge which doesn't require a name or ref
+     * tag. Will be ignored.
+     * 
+     * @param edge
+     *            edge in question
+     * @return boolean if edge is a small connector road that doesn't require a name or ref tag.
+     */
+    private boolean isConnectorWayToIgnore(final Edge edge)
```

Comment:
This feels like it may be creating a lot of false negative cases. I would be curious to see some examples of the type of Edge this is supposed to be filtering out.

Reply:
A connector road is a road that connects a split road like the following (Thought name/ref was optional on these OR they take higher priority name?):

![image](https://user-images.githubusercontent.com/18335251/112908425-5cbf2880-90a4-11eb-905c-3b83c1dd1429.png)

Reply:
Some have names and some don't, I will research proper modeling for these and get back.

Reply:
I see. This makes a decent bit of sense now. In a case like this I can see not needing a name, though I don't know the proper modeling either. The sort of case I would be worried about would be if the crossing road continued through the other side of the split road. In that case I think it should defiantly have a name or ref.

Reply:
addressed